frontend/src/metabase-lib/types.ts

```diff
@@ -131,7 +131,9 @@ export type MetricDisplayInfo = {
 export type ClauseDisplayInfo = Pick<
   ColumnDisplayInfo,
   "name" | "displayName" | "longDisplayName" | "table"
->;
+> & {
+  isNamed?: boolean;
+};
```

Comment:
Only expression, aggregation, and filter clauses are planned to have this property, but I didn't feel like adding it one by one or creating a new type. I can do that if that's important though.